app/models/miq_server/worker_management/monitor.rb

```diff
@@ -23,6 +23,9 @@ def monitor_workers
     # Sync the workers after sync'ing the child worker settings
     sync_workers
 
+    starting = MiqWorker.find_starting.find_all { |w| MiqWorkerType.worker_class_names.include?(w.class.name) }.to_a
```

Comment:
Can we move the `find_all` into a scope, maybe `MiqWorker.with_worker_type` or something? (maybe use this as well in on line 13)

@agrare Do you know of any reason why this (and the predecessor on line 13) are using a ruby `select / find_all` instead of sql? Maybe so we use consistent sql for looking up the starting workers?

e.g.:

```
class MiqWorker
  def self.with_worker_types
    where(:type => MiqWorkerType.worker_class_names)
  end
end
```

Reply:
Been like that "forever"

`7ef9b42c4ad vmdb/app/models/miq_server/worker_management/monitor/start.rb (Oleg Barenboim 2012-07-10 22:41:17 +0000 13)       starting = MiqWorker.find_starting.find_all { |w| self.class.monitor_class_names.include?(w.class.name) }`

Reply:
yea. I couldn't figure out why.